requirement-generator/requirement-satisfiable-course-generator.ts

```diff
@@ -17,25 +24,50 @@ function courseCanSatisfyCollegeOrMajorRequirement(
 
```

Comment:
Once the function is better flushed, we would want documentation for maintainability